### PR TITLE
[woptim] Switch CI to RADIUSS Shared CI

### PR DESCRIPTION
**Pull request by adrienbernede**
This branch updates the SAMRAI GitLab CI to RADIUSS Shared CI template.

### CODE REVIEW

**Comment by adrienbernede** [comment]
@keithhealy This is phase 1:
I moved to RADIUSS Shared CI, without any change to the builds themselves.
Some allocation issues to be fixed.

**Comment by adrienbernede** [comment]
When Tioga is back, we can trigger the CI again. Should be OK now.

File: .gitlab/jobs/tioga.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#
+# This file is part of the SAMRAI distribution.  For full copyright
+# information, see COPYRIGHT and LICENSE.
+#
+# Copyright:     (c) 1997-2023 Lawrence Livermore National Security, LLC
+#
+#########################################################################
+
+variables:
+  BUILD_OPTION: "--build-only"
```
Comment (nselliott) [inline .gitlab/jobs/tioga.yml:11]: It looks like this --build-only flag doesn't end up in the command, so the CI attempted to run the tests for this configuration that is supposed to be only a build.

**Comment by nselliott** [comment]
I made a branch `feature/nselliott/radiuss-shared-ci` that changes `tioga.yml` to use a newer configuration that should complete a full build and test, but something goes wrong when it tries to launch the pipeline.  @adrienbernede or @keithhealy could you check on that?  Feel free to merge that branch into this one if you can make it work.

**Review by nselliott — APPROVED** [review]

**Pull request merged**